plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/NullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    public NullPushdownDataTypeTest execute(QueryRunner queryRunner, Session session, DataSetup dataSetup)
+    {
+        checkState(!testCases.isEmpty(), "No test cases");
+        for (int specialColumn = 0; specialColumn < SPECIAL_COLUMNS; specialColumn++) {
+            checkArgument(!"NULL".equalsIgnoreCase(testCases.get(specialColumn).inputLiteral()));
+        }
+
+        try (TemporaryRelation temporaryRelation = dataSetup.setupTemporaryRelation(unmodifiableList(testCases))) {
+            verifyPredicate(queryRunner, session, temporaryRelation, true, false, !connectorExpressionOnly);
+            verifyPredicate(queryRunner, session, temporaryRelation, true, true, true);
+            verifyPredicate(queryRunner, session, temporaryRelation, false, false, !connectorExpressionOnly);
+            verifyPredicate(queryRunner, session, temporaryRelation, false, true, true);
+        }
+        return this;
+    }
+
+    private void verifyPredicate(QueryRunner queryRunner, Session session, TemporaryRelation temporaryRelation, boolean isNull, boolean connectorExpression, boolean expectPushdown)
+    {
+        TestCase firstCase = testCases.getFirst();
+        String firstColumnName = "col_0";
+        String withConnectorExpression = connectorExpression ? " OR %s IS NULL".formatted(firstColumnName) : "";
+
+        String queryWithAll = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " +
+                IntStream.range(SPECIAL_COLUMNS, testCases.size())
+                        .mapToObj(column -> getPredicate(column, isNull))
+                        .collect(joining(" AND "))
+                + withConnectorExpression;
+
+        // Closing QueryAssertions would close the QueryRunner
+        QueryAssertions queryAssertions = new QueryAssertions(queryRunner);
+        try {
+            assertPushdown(expectPushdown,
+                    assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                            assertThat(queryAssertions.query(session, queryWithAll))));
+        }
+        catch (AssertionError e) {
+            // if failed - identify exact column which caused the failure
+            for (int column = SPECIAL_COLUMNS; column < testCases.size(); column++) {
+                String queryWithSingleColumnPredicate = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " + getPredicate(column, isNull) + withConnectorExpression;
+                assertPushdown(expectPushdown,
+                        assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                                assertThat(queryAssertions.query(session, queryWithSingleColumnPredicate))));
+            }
+            throw new IllegalStateException("Single column assertion should fail for at least one column, if query of all column failed", e);
+        }
+    }
+
```

Comment:
Instead of asserting them twice like once with all columns `AND` ? Can we assert them each column at a time ? Like 

```
            for (int column = SPECIAL_COLUMNS; column < testCases.size(); column++) {
                String queryWithSingleColumnPredicate = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " + getPredicate(column, isNull) + withConnectorExpression;
                assertPushdown(expectPushdown,
                        assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
                                assertThat(queryAssertions.query(session, queryWithSingleColumnPredicate))));
            }
```

This would allow us to identify the specific columns which would cause failure in the future

Reply:
Not sure if I follow your suggestion. We already asserting each column at a time, in case of failure.

Reply:
Instead of handling twice can we try asserting only once for each column so that it would be easier to debug in case of failures

Reply:
The first check for all columns is N times faster than the second for single columns. So it's an improvement for sunny-day scenario. 
In case of failure there is overhead of 1 additional check, performing N additional checks for single columns.
This was copied from SqlDataTypeTest. 
Please let me know if you want to simplify code.

Reply:
Makes sense !! Let's have it like this.